engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/confluent/ConfluentSchemaRegistryProvider.scala

```diff
@@ -0,0 +1,81 @@
+package pl.touk.nussknacker.engine.avro.confluent
+
+import io.confluent.kafka.schemaregistry.client.{SchemaRegistryClient => ConfluentSchemaRegistryClient}
+import org.apache.flink.api.common.typeinfo.TypeInformation
+import pl.touk.nussknacker.engine.api.process.ProcessObjectDependencies
+import pl.touk.nussknacker.engine.avro.confluent.ConfluentSchemaRegistryClientFactory.TypedSchemaRegistryClient
+import pl.touk.nussknacker.engine.avro.confluent.formatter.ConfluentAvroToJsonFormatter
+import pl.touk.nussknacker.engine.avro.{SchemaRegistryClientFactory, SchemaRegistryProvider}
+import pl.touk.nussknacker.engine.kafka.serialization.{DeserializationSchemaFactory, SerializationSchemaFactory}
+import pl.touk.nussknacker.engine.kafka.{KafkaConfig, RecordFormatter}
+
+class ConfluentSchemaRegistryProvider[T: TypeInformation](val schemaRegistryClientFactory: SchemaRegistryClientFactory[ConfluentSchemaRegistryClient],
+                                                          val serializationSchemaFactory: SerializationSchemaFactory[Any],
+                                                          val deserializationSchemaFactory: DeserializationSchemaFactory[T],
+                                                          val kafkaConfig: KafkaConfig,
+                                                          val formatKey: Boolean) extends SchemaRegistryProvider[T] {
+  override def recordFormatter(topic: String): Option[RecordFormatter] =
+    Some(ConfluentAvroToJsonFormatter(createSchemaRegistryClient, topic, formatKey))
+
+  override def createSchemaRegistryClient: TypedSchemaRegistryClient =
+    schemaRegistryClientFactory.createSchemaRegistryClient(kafkaConfig)
+}
+
+object ConfluentSchemaRegistryProvider extends Serializable {
+
+  import net.ceedubs.ficus.Ficus._
+  import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+
+  def apply[T: TypeInformation](schemaRegistryClientFactory: SchemaRegistryClientFactory[ConfluentSchemaRegistryClient],
```

Comment:
do we really need all those factory methods?

Reply:
Left only 3.